src/Illuminate/Database/Console/Migrations/StatusCommand.php

```diff
@@ -75,14 +75,23 @@ public function handle()
      */
     protected function getStatusFor(array $ran, array $batches)
     {
-        return Collection::make($this->getAllMigrationFiles())
-                    ->map(function ($migration) use ($ran, $batches) {
-                        $migrationName = $this->migrator->getMigrationName($migration);
-
-                        return in_array($migrationName, $ran)
-                                ? ['<info>Y</info>', $migrationName, $batches[$migrationName]]
-                                : ['<fg=red>N</fg=red>', $migrationName];
-                    });
+        $migrations = [];
+
+        foreach ($batches as $migration => $batch) {
+            $migrations[] = ['<info>Y</info>', $migration, $batch];
+        }
+
+        // Diff to find ones that aren't ran
+        $diff = Collection::make($this->getAllMigrationFiles())
+            ->map(function ($migration) {
+                return $this->migrator->getMigrationName($migration);
+            })->diff($ran);
+
+        foreach ($diff as $migrationName) {
```

Comment:
Why the foreach, and not just another map?

Reply:
True. I'll replace the foreach's with collections